File: elasticmeasures/Dep/dtw_ip.py
import numpy as np

def dtw_ip(x,y,slope = 5):

    cur = np.full(len(y),np.inf);
    prev = np.full(len(y),np.inf);
    xlen = len(x);
    ylen = len(y);

    min_slope = (1/slope) * (float(ylen)/float(xlen));
    max_slope = slope * (float(ylen)/float(xlen));


    for i in range(len(x)):
        temp = prev;
        prev = cur;
        cur = temp;
        minw =  np.ceil(max(min_slope * i,
                    ((ylen-1) - max_slope * (xlen - 1)
                        + max_slope * i)))
        maxw = np.floor(min(max_slope * i,
                   ((ylen - 1) - min_slope * (xlen - 1)
                      + min_slope * i)) + 1);
        for j in range(int(minw),int(maxw)):

            if i + j == 0:
                cur[j] = abs(x[0] - y[0]) ** 2;
            elif i == 0:
                cur[j] = abs(x[0] - y[j]) ** 2 + cur[j-1];
            elif j == 0:
                cur[j] = abs(x[i] - y[0]) ** 2 + prev[j];
            else:
                cur[j] = abs(x[i] - y[j]) ** 2 + min(prev[j-1],prev[j],cur[j-1]);
    final_dtw = cur[len(y)-1];

    return final_dtw ** (1/2);

File: elasticmeasures/Dep/dtw_ip.cpp
#include <stdlib.h>
#include <stdio.h>
#include <math.h>

#define min(x, y) ((x)<(y)?(x):(y))
#define max(x, y) ((x)>(y)?(x):(y))
#define dist(x, y)((x-y)*(x-y))

double dtw_ip(double* x, int xlen, double* y, int ylen, int w)
{
	double* prev, * cur, * temp;
	cur = (double*)malloc(ylen * sizeof(double));
	prev = (double*)malloc(ylen * sizeof(double));
	int i, j;
	double s1, s2, s3;

	double min_slope = 1 / s * (double)xlen / (double)ylen;
	double max_slope = s * (double)xlen / (double)ylen;

	min_slope* i;
	(double)(((double)ylen - 1) - max_slope * ((double)xlen - 1))
		+ max_slope * i;

	max_slope* i;
	(((double)ylen - 1) - min_slope * ((double)xlen - 1))
		+ min_slope * i;

	for (i = 0; i < xlen; i++)
	{
		temp = prev;
		prev = cur;
		cur = temp;
		int minw = max(min_slope * i,
			(double)(((double)ylen - 1) - max_slope * ((double)xlen - 1))
			+ max_slope * i);
		int maxw = min(max_slope * i,
			(((double)ylen - 1) - min_slope * ((double)xlen - 1))
			+ min_slope * i);
		for (j = minw; j < maxw; j++)
		{
			if (i + j == 0)
			{
				cur[j] = dist(x[i], y[j]);
			}
			else if (i == 0)
			{
				cur[j] = dist(x[0], y[j]) + cur[j - 1];
			}
			else if (j == 0)
			{
				cur[j] = dist(x[i], y[0]) + prev[j];
			}
			else
			{
				cur[j] = dist(x[i], y[j]) + min(prev[j - 1], min(prev[j], cur[j - 1]));
			}
		}


	}

	double result = cur[ylen - 1];
	free(cur);
	free(prev);

	return result;
}